tony-core/src/main/java/com/linkedin/tony/events/EventHandler.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.tony.events;
+
+import com.linkedin.tony.TonyJobMetadata;
+import com.linkedin.tony.util.HistoryFileUtils;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.concurrent.BlockingQueue;
+import org.apache.avro.file.DataFileWriter;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.Path;
+
+
+public class EventHandler implements Runnable {
+  private static final Log LOG = LogFactory.getLog(EventHandler.class);
+  private boolean isStopped = false;
+  private BlockingQueue<Event> eventQueue;
+  private Path historyFile = null;
+  private Path tmpFile = new Path("tmp.avro");
+  private DatumWriter<Event> eventWriter = new SpecificDatumWriter<>();
+  private DataFileWriter<Event> dataFileWriter = new DataFileWriter<>(eventWriter);
+  private OutputStream out;
+  private FileSystem myFs;
+
+  public EventHandler(FileSystem fs, BlockingQueue<Event> q) {
+    eventQueue = q;
+    myFs = fs;
+    try {
+      out = myFs.create(tmpFile);
+      dataFileWriter.create(Event.SCHEMA$, out);
+    } catch (IOException e) {
+      LOG.error("Failed to set up writer", e);
+    }
+  }
+
+  public void emitEvent(Event event) {
+    try {
+      eventQueue.put(event);
+    } catch (InterruptedException e) {
+      LOG.error("Failed to add event " + event + " to event queue", e);
+    }
+  }
+
+  private void writeEvent(BlockingQueue<Event> queue, DataFileWriter<Event> writer) {
```

Comment:
I think you don't need to pass `queue` and `writer` into this method since they are instance variables already.

Reply:
I just think it would be easier to write test this way.